docs/remote/devcontainerjson-reference.md

```diff
@@ -59,6 +59,12 @@ While some devcontainer.json properties apply generally, others are only used in
 
 If you've already built the container and connected to it, be sure to run **Remote-Containers: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up the change.
 
+> **Note:** The format of certain properties will vary depending on the involvement of a shell.
+
+> `postCreateCommand`, `postStartCommand`, `postAttachCommand`, and `initializeCommand` all have an array and a string type. An array is passed to the OS for execution without going through a shell, whereas a string goes through a shell (it needs to be parsed into command and arguments).
+
+> When using `runArgs` via a typical command line, you'll need single quotes if the shell runs into parameters with spaces. However, these single quotes aren't passed on to the executable. Thus, in your `devcontainer.json`, you'd follow the array format used by the properties mentioned above, i.e.: `"runArgs": ["--device-cgroup-rule=my rule here"]`, rather than `"runArgs": ["--device-cgroup-rule='my rule here'"]`. Its arguments will be passed as an array, and no quoting for any shell is involved.
```

Comment:
The name "runArgs" doesn't appear on the command line, I'd rephrase "using `runArgs` via a typical command line". Or: Use an example with "post*Command" comparing the string and the array versions: `"postAttachCommand": "echo foo='bar'"` will remove the single quotes as part of the shell's parsing whereas `"postAttachCommand": ["echo", "foo='bar'"]` will keep them and write them to standard out. (You can see the output in the Dev Container log.)

Reply:
These are great suggestions too. I rephrased using your first suggestion and also added the `postAttachCommand` example. Let me know what you think!